common/src/indexer_service/http/request_handler.rs

```diff
@@ -91,20 +96,14 @@ where
         (true, Some(signer)) => {
             let req = serde_json::to_string(&request)
                 .map_err(|_| IndexerServiceError::FailedToSignAttestation)?;
-            let res = serde_json::to_string(&response)
+            let res = response
+                .as_str()
                 .map_err(|_| IndexerServiceError::FailedToSignAttestation)?;
             Some(signer.create_attestation(&req, &res))
         }
     };
 
-    let mut headers = HeaderMap::new();
-    if let Some(attestation) = attestation {
-        let raw_attestation = serde_json::to_string(&attestation)
-            .map_err(|_| IndexerServiceError::FailedToProvideAttestation)?;
-        let header_value = HeaderValue::from_str(&raw_attestation)
-            .map_err(|_| IndexerServiceError::FailedToProvideAttestation)?;
-        headers.insert("graph-attestation", header_value);
-    }
+    let response = response.finalize(attestation);
 
-    Ok((StatusCode::OK, headers, response))
+    Ok((StatusCode::OK, response))
```

Comment:
Yeah, the header was a mistake. What the gateway expects right now is a JSON response in the form
```json
{
  "graphQLResponse": "...",
  "attestation": null // or a signature struct
}
```
I'm not sure I like it that way, headers would be more universal - not every data service will return JSON, and moving the attestation into the body is what required splitting up processing the request and finalizing the response in the service implementation. It would've been nice to not require `finalize` and instead add the attestation into the headers like the initial PR for the indexer service framework did.

But... we can change this later, it's not a big deal. For now I think we care more about compatibility with the gateway.

Reply:
Yup, thanks for info!